python/ray/train/v2/_internal/execution/worker_group/worker_group.py

```diff
@@ -233,6 +260,11 @@ def _start_impl(
             for callback in self._callbacks:
                 callback.before_worker_group_start(worker_group_context)
 
+            self._check_cluster_resources_and_raise_if_insufficient(
```

Comment:
Should we move this up to the top of `_start_impl`? Right now this happens after `before_worker_group_start`, which creates a bunch of `TrainRun` objects that show up in the dashboard.